FieldOpt/Model/model.h

```diff
@@ -121,6 +125,22 @@ class Model : public Loggable
    private:
     Model *model_;
   };
+ public:
+  class economy{
```

Comment:
* Class and struct names should have Capital First Letter
* IMO this should be a `struct`, not a `class`. A struct is typically used when all member attributes are public (directly accessible) (and when the object acts primarily as a container and doesn't to much else by itself)